### PR TITLE
Migrate triple equals rule to 1.4

### PR DESCRIPTION
**Pull request by adidahiya**

### CODE REVIEW

File: src/rules/tripleEqualsRule.ts [hunk 1]
```diff
@@ -22,44 +22,38 @@ export class Rule extends Lint.Rules.AbstractRule {
...
 class ComparisonWalker extends Lint.RuleWalker {
-    public visitBinaryExpression(node: TypeScript.BinaryExpressionSyntax): void {
-        var position = this.positionAfter(node.left);
+    private static COMPARISON_OPERATOR_WIDTH = 2;
 
+    public visitBinaryExpression(node: ts.BinaryExpression) {
         if (!this.isExpressionAllowed(node)) {
-            this.handleOperatorToken(position, node.operatorToken);
+            var endOfLeftExpression = node.left.getEnd();
+            this.handleOperatorToken(endOfLeftExpression + 1, node.operator);
```
Comment (adidahiya) [inline src/rules/tripleEqualsRule.ts]: this is a bit of a hack, but there's not a whole lot we can do given that `node.operator` no longer has position info, it's just a `SyntaxKind`

**Comment by ashwinr** [comment]
Please resolve the conflicts and update the PR so I can merge it

File: src/rules/tripleEqualsRule.ts (hunk 1 above)
Comment (ashwinr) [inline src/rules/tripleEqualsRule.ts]: Yeah I saw that. It's fine.

Sent from a phone

On Jan 21, 2015, at 20:38, Adi Dahiya <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/rules/tripleEqualsRule.ts<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_tslint_pull_264-23discussion-5Fr23350245&d=AwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=jafYnmwAIy7b95Z05TKJr1hFAxZozsv6_JDDlWyuJk8&m=H3SKsTsUIHcSrcBRFQuU9RzhK42iaVYEy6BqR-JHEQc&s=H2sXpwHt6jV-s0z4nJY62vWB8P1CkHhNG1sRRCefQKw&e=>:

>          if (!this.isExpressionAllowed(node)) {
> -            this.handleOperatorToken(position, node.operatorToken);
> +            var endOfLeftExpression = node.left.getEnd();
> +            this.handleOperatorToken(endOfLeftExpression + 1, node.operator);


this is a bit of a hack, but there's not a whole lot we can do given that node.operator no longer has position info, it's just a SyntaxKind

Comment (gscshoyru) [inline src/rules/tripleEqualsRule.ts]: Actually you can use .getChild(1) to get the node for the operator.

Comment (adidahiya) [inline src/rules/tripleEqualsRule.ts]: @gscshoyru sweet, I'll try that out

**Pull request merged**